### ISSUE TITLE
Connection through Tor/Socks5 doesn't work

### ISSUE THREAD

**Issue by Revertron** [yggdrasil-network/yggdrasil-go#1028]
If you try to add something like `socks://127.0.0.1:9050/ygg.mkg20001.io:80` into peers you will get `invalid username/password` in yggdrasil log.

But if I configure some username and password in Tor and use those creds in socks:// url it works.
I think Yggdrasil needs to send empty username+password to the socks-server.

**Comment by majestrate** [comment]
if tor does not support no auth in socks5 would it not be a bug in our socks5 client dep? it should send back an error like "proxy refused request, authorization type [placeholder values goes here] not supported" if we are talking to a socks5 proxy that does not accept anon socks5 request.

**Comment by Revertron** [comment]
But it works from Firefox without credentials. Should work from Yggdrasil, no?

**Comment by Vort** [comment]
Maybe you want to try setting `SocksPort 9050 PreferSOCKSNoAuth` in `torrc`.

**Comment by 0Flag** [comment]
Also I2P connections stopped working with error: `unknown error host unreachable`. Tor and I2P connections worked fine for 0.4.4. It seems there was a refactoring from 0.4.4 to 0.4.5 which borked the socks connections.

**Comment by Vort** [comment]
I2P network is under attack now by the way. Not sure if it is related to proxy problem however.

**Comment by 0Flag** [comment]
> I2P network is under attack now by the way. Not sure if it is related to proxy problem however.
True, but that's not the issue. I can connect to I2P peers right now with V 0.4.4

**Comment by neilalexander** [comment]
Is this still an issue?

**Comment by r4sas** [comment]
> Is this still an issue?

Yes.
